overlays/crypto/libsecp256k1.nix

```diff
@@ -11,6 +11,8 @@ stdenv.mkDerivation rec {
   configureFlags = [
     "--enable-benchmark=no"
     "--enable-module-recovery"
+  ] ++ lib.optional stdenv.hostPlatform.isMusl [
+    "--enable-static"
   ];
 
```

Comment:
I don't think this is good. We have many more customisations in haskell.nix, that I'm not sure should be in iohk-nix? For `musl` I can see this being on ok.

Reply:
well, if we don't want to override nixpkgs `secp256k1` , then we need to implement here the customizations of `secp256k1` done by haskell.nix musl overlay.

Reply:
@iquerejeta mentioned that we want to pin our crypto libs to insulate us from upstream changes and have better control over the exact versions.

Reply:
I meant `haskell.nix` is adding `"--enable-static"` to `secp256k1`, but if we use a different name (`libsecp256k1`) then we need to manage `"--enable-static"` in iohk-nix, right?